packages/components/constraints/src/horizontal/horizontal.js

```diff
@@ -1,63 +1,91 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { css } from '@emotion/core';
+import invariant from 'tiny-invariant';
 import { filterDataAttributes } from '@commercetools-uikit/utils';
 import { customProperties as vars } from '@commercetools-uikit/design-system';
 
-const getConstraintSyles = (constraint) => {
-  switch (constraint) {
-    case 'xs':
-      return css`
-        max-width: ${vars.constraintXs};
-      `;
-    case 's':
-      return css`
-        max-width: ${vars.constraintS};
-      `;
-    case 'm':
-      return css`
-        max-width: ${vars.constraintM};
-      `;
-    case 'l':
-      return css`
-        max-width: ${vars.constraintL};
-      `;
-    case 'xl':
-      return css`
-        max-width: ${vars.constraintXl};
-      `;
-    case 'scale':
-      return css`
-        width: ${vars.constraintScale};
-      `;
-    default:
-      return css``;
-  }
-};
+function getConstraintSyles({ max, constraint }) {
+  let constraintToken;
+
+  if (max === 1 || constraint === 'xs') constraintToken = vars.constraint1;
+  if (max === 2) constraintToken = vars.constraint2;
+  if (max === 3 || constraint === 's') constraintToken = vars.constraint3;
+  if (max === 4) constraintToken = vars.constraint4;
+  if (max === 5) constraintToken = vars.constraint5;
+  if (max === 6) constraintToken = vars.constraint6;
+  if (max === 7 || constraint === 'm') constraintToken = vars.constraint7;
+  if (max === 8) constraintToken = vars.constraint8;
+  if (max === 9) constraintToken = vars.constraint9;
+  if (max === 10 || constraint === 'l') constraintToken = vars.constraint10;
+  if (max === 12) constraintToken = vars.constraint12;
+  if (max === 13) constraintToken = vars.constraint13;
+  if (max === 14) constraintToken = vars.constraint14;
+  if (max === 15) constraintToken = vars.constraint15;
+  if (max === 16 || constraint === 'xl') constraintToken = vars.constraint16;
+
+  return css`
+    ${constraintToken ? `max-width: ${constraintToken};` : ''}
+    ${max === 'auto' ? 'width: auto;' : ''}
+  `;
+}
 
-const Horizontal = (props) => (
-  <div
-    css={[
-      css`
-        width: 100%;
-        position: relative;
-      `,
-      getConstraintSyles(props.constraint),
-    ]}
-    {...filterDataAttributes(props)}
-  >
-    {props.children}
-  </div>
-);
+const Horizontal = (props) => {
+  invariant(
+    props.constraint && props.max,
+    '`ui-kit/constraints/horizontal: props `constraint` and `max` should not be used in conjunction. Please prefer `max` prop.'
+  );
+
+  return (
+    <div
+      css={[
+        css`
+          width: 100%;
+          position: relative;
+        `,
+        getConstraintSyles(props),
+      ]}
+      {...filterDataAttributes(props)}
+    >
+      {props.children}
+    </div>
+  );
+};
 
 Horizontal.displayName = 'Horizontal';
 Horizontal.propTypes = {
+  /**
+   * Determines scale of the constraint.
+   */
+  max: PropTypes.oneOf([
+    1,
```

Comment:
this looks fragile. can this be

```js
const maxProp = [...Array.from({ length: 16 }).map((_, index) => index + 1), 'scale', 'auto']
// [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,"scale","auto"]
```

Reply:
I tried it but unfortunately, it messes up the `generate-readme` script 😞